advanced_function/task3.js

```diff
@@ -0,0 +1,16 @@
+function divide(numerator, denominator) {
+    if (typeof numerator !== 'number' || typeof denominator !== 'number' || denominator === 0) {
+        throw new Error("Invalid data: the denominator can't be zero & both arguments should be numeric");
```

Comment:
it would be better use dedicated `if` blocks for each validation rule. it would allow us to provide better message about error